heat/core/tensor.py

```diff
@@ -13,6 +14,22 @@
 from . import rounding
 from . import reductions
 
+warnings.simplefilter('always', ResourceWarning)
```

Comment:
This seems like a very strong filter.

Why is this here? Can we filter this warning precisely where it happens, e.g. as a decorator around the function that produces the warning?